src/ServiceControl/Notifications/Api/NotificationsController.cs

```diff
@@ -0,0 +1,135 @@
+
+                return new JsonResult<EmailNotifications>(
+                    settings.Email,
+                    new JsonSerializerSettings
+                    {
+                        NullValueHandling = NullValueHandling.Include,
+                        ContractResolver = new UnderscoreMappingResolver()
+                    },
+                    Encoding.Unicode,
+                    request);
+            }
+        }
+
+        [Route("notifications/email/toggle")]
+        [HttpPost]
+        public async Task<HttpResponseMessage> ToggleEmailNotifications(ToggleEmailNotifications request)
+        {
+            using (var session = store.OpenAsyncSession())
+            {
+                var settings = await LoadSettings(session).ConfigureAwait(false);
+
+                settings.Email.Enabled = request.Enabled;
+
+                await session.SaveChangesAsync().ConfigureAwait(false);
+            }
+
+            return new HttpResponseMessage(HttpStatusCode.OK);
+        }
+
+        [Route("notifications/email")]
+        [HttpPost]
+        public async Task<HttpResponseMessage> UpdateSettings(UpdateEmailNotificationsSettingsRequest request)
+        {
+            using (var session = store.OpenAsyncSession())
+            {
+                var settings = await LoadSettings(session).ConfigureAwait(false);
+
+                var emailSettings = settings.Email;
+
+                emailSettings.SmtpServer = request.SmtpServer;
+                emailSettings.SmtpPort = request.SmtpPort;
+
+                emailSettings.AuthenticationAccount = request.AuthorizationAccount;
+                emailSettings.AuthenticationPassword = request.AuthorizationPassword;
+                emailSettings.EnableTLS = request.EnableTLS;
+
+                emailSettings.From = request.From;
+                emailSettings.To = request.To;
+
+                await session.SaveChangesAsync().ConfigureAwait(false);
+
+                return new HttpResponseMessage(HttpStatusCode.OK);
+            }
+        }
+
+        [Route("notifications/email/test")]
+        [HttpPost]
+        public async Task<HttpResponseMessage> SendTestEmail()
+        {
+            using (var session = store.OpenAsyncSession())
+            {
+                var settings = await LoadSettings(session).ConfigureAwait(false);
+
+                try
+                {
+                    await EmailSender.Send(
+                            settings.Email,
+                            $"[{instanceName}] health check notification check successful",
+                            $"[{instanceName}] health check notification check successful.")
+                        .ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    return new HttpResponseMessage(HttpStatusCode.InternalServerError)
+                    {
+                        Content = new StringContent($"{e.Message} {e.InnerException?.Message}"),
```

Comment:
Is it ever a good idea to use the exception message and expose that to the user?

Reply:
These are details that are shown to the user testing SMTP connectivity and include pretty descriptive information like "Could not reach server" or "Server certificate is not valid" etc.